client/app/components/dashboards/CreateDashboardDialog.jsx

```diff
@@ -0,0 +1,86 @@
+import { trim } from 'lodash';
+import React, { useRef, useState, useEffect } from 'react';
+import Modal from 'antd/lib/modal';
+import Input from 'antd/lib/input';
+import DynamicComponent from '@/components/DynamicComponent';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import { $location, $http } from '@/services/ng';
+import recordEvent from '@/services/recordEvent';
+import { policy } from '@/services/policy';
+
+function CreateDashboardDialog({ dialog }) {
+  const [name, setName] = useState('');
+  const [isValid, setIsValid] = useState(false);
+  const [saveInProgress, setSaveInProgress] = useState(false);
+  const inputRef = useRef();
+  const isCreateDashboardEnabled = policy.isCreateDashboardEnabled();
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    }, 100);
+    return () => clearTimeout(timer);
+  }, []);
```

Comment:
I think you can substitute this for `<Input autofocus />`, no?

Reply:
It does not work in modals. This fix was suggested in [related discussion](https://github.com/ant-design/ant-design/issues/8668)

Reply:
Weird, it seems to work on `CreateSourceDialog` and `ChangePasswordDialog` :thinking:

Reply:
@gabrieldutra I'll check once more

Reply:
@gabrieldutra I see that `autoFocus` work in other modals, but it doesn't work in this one - I really have no idea what's wrong :thinking: I tried to simplify that dialog as mu as i can - removed almost all attributes, `<DynamicComponent>`, etc. - just `<Modal ...><Input autoFocus /></Modal>` - input does not get focused.

Reply:
There's one significant difference - Class component vs Functional component 🤔

Reply:
@ranbena Interesting idea. I'll try to implement it as class component - will see what happens.

Reply:
@ranbena Tried class component - the same issue (no autofocus).

Upd.: When this dialog is opened from React code - Input gets autofocus, when from Angular - no. I'll add a comment with `ANGULAR_REMOVE_ME`

Reply:
> When this dialog is opened from React code - Input gets autofocus, when from Angular - no.

Makes sense and also good to know this sort of thing happens :thinking:

Reply:
Yeah... But usually applications are built using only one framework, so such shit does not happen :slightly_smiling_face: